configure.ac

```diff
@@ -0,0 +1,191 @@
+
+AC_SEARCH_LIBS([uncompress], [z])
+
+# Enable systemd per option by setting HAVE_SYSTEMD
+
+AS_CASE([${enable_systemd}],
+  [yes|check],
+    [AC_MSG_CHECKING([systemd])
+     AC_SEARCH_LIBS([sd_listen_fds], [systemd-daemon])
+     AC_CHECK_FUNC([sd_listen_fds],
+       [AC_MSG_RESULT([enabled: systemd])
+        AC_DEFINE([HAVE_SYSTEMD],[1],
+          [Define to 1 to enable systemd support])],
+       AS_CASE([${enable_systemd}],
+         [yes],
+           AC_MSG_ERROR([systemd support not available]),
+	 AC_MSG_RESULT([disabled: systemd])))],
+  [no],,
+  [AC_MSG_ERROR([invalid argument, --enable-systemd=${enable_systemd}])])
+
+# Enable selinux per option by setting HAVE_SETCON
+
+AS_CASE([${enable_selinux}],
+  [yes|check],
+    [AC_SEARCH_LIBS([setcon], [selinux])
+     AC_CHECK_FUNCS([setcon],
+       AC_MSG_RESULT([enabled: selinux]),
+       AS_CASE([${enable_selinux}],
+         [yes],
+           AC_MSG_ERROR([selinux support not available]),
+         [check],
+           AC_MSG_RESULT([disabled: selinux])))],
+  [no],
+    AC_MSG_RESULT([disabled: selinux]),
+  [AC_MSG_ERROR([invalid argument, --enable-selinux=${enable_selinux}])])
+
+# The check library is used for the unit testing.  Not bothering to
+# implement an --enable-tests option, as they are only run by "make
+# check" so there is no user-level difference.  If the unit tests
+# won't even *compile* there is probably a better solution than just
+# turning them off, and build time is minimal.
+
+PKG_CHECK_MODULES(CHECK,[check],
+  AC_MSG_RESULT([enabled: unit testing]),
+  AC_MSG_RESULT([disabled: unit testing]))
+
+AM_CONDITIONAL([HAVE_CHECK],[test -z "${CHECK_PKG_ERRORS}"])
+
+# For each commented AC_SEARCH_LIBS below, uncomment and replace ? by
+# called procedure provided only by that library on the platform(s) in
+# question.
+
+# for SunOS / solaris:
+#AC_SEARCH_LIBS([?], [nsl])
+
+# for BeOS:
+#AC_SEARCH_LIBS([?], [bind])
+#AC_SEARCH_LIBS([?], [bsd])
+
+# for SunOS / solaris / BeOS:
+#AC_SEARCH_LIBS([?], [socket])
+
+# for Haiku:
+#AC_SEARCH_LIBS([?], [network])
+
+# for MS windows32:
+#AC_SEARCH_LIBS([?], [ws2_32])
+#AC_SEARCH_LIBS([?], [iphlpapi])
+
+# Checks for headers
+
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h netdb.h netinet/in.h stdint.h stdlib.h string.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h syslog.h termios.h unistd.h systemd/sd-daemon.h selinux/selinux.h])
+
+# Checks for typedefs, structures, and compiler characteristics
+
+AC_CHECK_TYPES([socklen_t],,
+  AC_DEFINE([socklen_t],[int],
+    [define to int if type socklen_t not found]),
+  [
+#ifdef HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
+#ifdef HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#endif
+])
+
+AC_TYPE_UID_T
+AC_C_INLINE
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions
+
+AC_FUNC_FORK
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([alarm dup2 inet_ntoa memset select socket strcasecmp strchr strdup strerror strrchr strstr])
```

Comment:
Agreed. Go for it! Most of the checks are not really sensible, and the #ifdefs in the code proper could be normalized to use the HAVE_xxx properties generated by autoconf almost exclusively.

(I used autoscan and plugged along from there pretty brainlessly.)